source/_integrations/switchbot.markdown

```diff
@@ -67,7 +71,44 @@ There are three attributes available on the SwitchBot entity to give you more in
 
 - `Retry count`: How many times to retry sending commands to your SwitchBot devices.
 
-### Error codes and troubleshooting
+## SwitchBot Lock
+
+The integration currently only uses the primary lock state; in dual lock mode, not all things might work properly.
+
+A SwitchBot lock can be set up in Home Assistant in two different ways. You can enter the key id and encryption key yourself, or Home Assistant can import them from your SwitchBot account.
+### SwitchBot account (recommended)
+Using this option you can provide your SwitchBot account login credentials and Home Assistant will import the appropriate encryption key from your account.
+
+{% configuration_basic %}
+Username:
+  description: SwitchBot account username
+Password:
+  description: SwitchBot account password
+{% endconfiguration_basic %}
+
+<div class='note warning'>
+This integration doesn't support SSO accounts (Login with Google, etc.) only username and password accounts.
+</div>
+
+### Enter lock encryption key manually
```

Comment:
I think we should move this to the pySwitchbot repo readme and link it so any changes to the procedure are always up to date if we change any of the script params etc

Reply:
Pushed an update doing just that, I will make a PR for the library a little bit later.